vllm/v1/engine/async_llm.py

```diff
@@ -439,17 +447,37 @@ async def generate(
                 # own request cleanup based on finished.
                 finished = out.finished
                 assert isinstance(out, RequestOutput)
+                if (
+                    len(out.outputs) > 0
+                    and out.outputs[0].stop_reason == "close_streaming_session"
+                ):
+                    return
                 yield out
 
         # If the request is disconnected by the client, generate()
         # is cancelled or the generator is garbage collected. So,
         # we abort the request if we end up here.
-        except (asyncio.CancelledError, GeneratorExit):
+        except asyncio.CancelledError:
             await self.abort(request_id)
             if self.log_requests:
                 logger.info("Request %s aborted.", request_id)
             raise
 
+        except GeneratorExit:
+            if close_streaming_session is not None:
+                if self.log_requests:
+                    logger.info(
+                        "Request %s generator completed, session remains alive.",
+                        request_id,
+                    )
+                # For streaming sessions, generator completion is normal
+                return
```

Comment:
Can you elaborate a bit for the behavior here?

Reply:
This change wouldn't be needed with the alternative proposed API